### ISSUE TITLE
Incompatibility with Gravity Forms plugin

### ISSUE THREAD

**Issue by GoogleCodeExporter** [viktorthang/wordpress-custom-content-type-manager#155]
```
What steps will reproduce the problem?
1. Activate (on main site of multisite install)


What is the expected output? What do you see instead?

"Plugin could not be activated because it triggered a fatal error.

Fatal error: Cannot redeclare class FormElement in 
/home/mysite/public_html/wp-content/plugins/custom-content-type-manager/includes
/FormElement.php on line 29"


Does the problem continue if you disable all other plugins?
Haven't tried yet. No time for that now.

What version of the plugin are you using?
0.9.3.3

What version of WordPress?
3.2.1 multisite

Please provide any additional information below (e.g. Browser, version,
OS):
Firefox 3.6
```

Original issue reported on code.google.com by `milehigh...@gmail.com` on 30 Aug 2011 at 7:52

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the bug report.  This type of thing is extremely difficult to 
troubleshoot, thus the specific question about disabling other plugins (i.e. if 
you don't have time to verify it, then I have no way of fixing it).  It sounds 
like you have another plugin that uses a class named "FormElement" that is 
conflicting with the CCTM's class of the same name.  There's no easy way around 
this, it's just the realities of PHP namespaces.  

I have built in several checks in my code to alert the admin to this type of 
condition, so I'll add a check for any existing classes named "FormElement", 
but all that does for you is it makes the thing fail gracefully, it doesn't 
make it work.  

If you require an immediate solution, try doing a find-and-replace across all 
files in the plugin and replace any instance of "FormElement" with an alternate 
name, e.g. "CCTMFormElement".  I may incorporate this classname into future 
versions of the plugin, but again, without verification that this is in fact 
the source of the problem, I unfortunately cannot take action on this issue.
```

Original comment by `fireproofsocks` on 30 Aug 2011 at 10:13
- Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
Thanks. Sounds reasonable.

I don't really want to update the plugin code since that would make
upgrading and maintenance difficult.

If you get around to changing the namespace I'd like to be notified if that
is possible.

FYI, I'm using Gravity Forms plugin if that helps.


Thanks,
~Jeff
*
Jeff Kemp, Founder/Key Executive *
*SiteSubscribe.com* <http://SiteSubscribe.com>
*MileHighMarketing* <http://MileHighMarketing.com>
*
Ph: (303) 888-5140*

<http://SiteSubscribe.com>

*SiteSubscribe.com >> Subscription Business Websites


<http://MileHighMarketing.com>
Higher Visibility <http://MileHighMarketing.com>**
*

<http://MileHighTechGuy.com>
*MileHighTechGuy.com*

jeff@MileHighTechGuy.com



On Tue, Aug 30, 2011 at 4:14 PM, <
wordpress-custom-content-type-manager@googlecode.com> wrote:
```

Original comment by `milehigh...@gmail.com` on 30 Aug 2011 at 10:26

**Comment by GoogleCodeExporter** [comment]
```
What's the URL for the Gravity Forms plugin?  I can check to see if they're 
using that classname.  It's always cat-and-mouse with that stuff, and PHP 
doesn't make it easy.  Sign up to the mailing list if you want to be notified 
of the upcoming version: http://eepurl.com/dlfHg

I'm working on a lot of features/fixes for 0.9.4, and if I can verify the the 
classname, I can change it easily enough.  Then upgrading wouldn't be an issue 
for you if you did make custom changes.
```

Original comment by `fireproofsocks` on 30 Aug 2011 at 10:52

**Comment by GoogleCodeExporter** [comment]
```
http://www.gravityforms.com/

Here's one location of it on my site:
http://sitesubscribe.com/contact/


Thanks,
~Jeff
*
Jeff Kemp, Founder/Key Executive *
*SiteSubscribe.com* <http://SiteSubscribe.com>
*MileHighMarketing* <http://MileHighMarketing.com>
*
Ph: (303) 888-5140*

<http://SiteSubscribe.com>

*SiteSubscribe.com >> Subscription Business Websites


<http://MileHighMarketing.com>
Higher Visibility <http://MileHighMarketing.com>**
*

<http://MileHighTechGuy.com>
*MileHighTechGuy.com*

jeff@MileHighTechGuy.com



On Tue, Aug 30, 2011 at 4:53 PM, <
wordpress-custom-content-type-manager@googlecode.com> wrote:
```

Original comment by `milehigh...@gmail.com` on 30 Aug 2011 at 11:03

**Comment by GoogleCodeExporter** [comment]
```
I'm changing this to an incompatibility issue, and I've already fixed it 
locally in my dev branch -- this will be included in version 0.9.4, but since 
I'm deep in the middle of some big code changes, I can't commit the dev version 
immediately, but it'll be available soon.
```

Original comment by `fireproofsocks` on 30 Aug 2011 at 11:29
- Changed title: **Incompatibility with Gravity Forms plugin**
- Changed state: **Fixed**
